relay-server/src/actors/project.rs

```diff
@@ -600,6 +616,63 @@ impl Actor for Project {
     }
 }
 
+struct RateLimitEnvelope {
+    event_id: Option<EventId>,
+    remote_addr: Option<IpAddr>,
```

Comment:
You could move these two into `EnvelopeSummary`. It looks like this type is a utility that would fit well into `utils::rate_limits`